src/main/java/me/earth/earthhack/impl/modules/render/popchams/PopChams.java

```diff
@@ -40,18 +44,12 @@ public PopChams()
         this.unregister(super.height);
     }
 
-    @Override
-    protected void onEnable()
-    {
-        popDataHashMap.clear();
-    }
-
-    public HashMap<String, PopData> getPopDataHashMap() {
-        return popDataHashMap;
+    public List<PopData> getPopDataList() {
+        return popDataList;
     }
 
     protected Color getColor(EntityPlayer entity) {
-        if (entity == mc.player) {
+        if (entity.getName().equals(entity.getName())) {
```

Comment:
```java
entity == mc.player
```
can we get that back? The new check doesnt really make sense the way I see it.

Reply:
Or well with the PlayerCopy set the EntityId of the PlayerCopy to that of the Entity and then use equals

Reply:
Cause rn all entities get the selfColor

Reply:
oh yeah ok ill fix that

Reply:
> set the EntityId of the PlayerCopy to that of the Entity and then use equals

pls like that, I think thats the best way, equals on entities only compares the entityId so that should work